### PR TITLE
Added Custom Input Box Toggle Feature

### PR DESCRIPTION
**Pull request by astu9880**

### CODE REVIEW

**Comment by astu9880** [comment]
@championswimmer @omerjerk @prateek27  Please review.

**Comment by astu9880** [comment]
Fixes #50

**Comment by hereisnaman** [comment]
@championswimmer this dont have option to close the input box again. I have included it in: https://github.com/coding-blocks/ide/pull/52

**Comment by astu9880** [comment]
@championswimmer it does. clicking on the custom input button toggles the state of custom input

**Comment by hereisnaman** [comment]
@astu9880 then it should also update the button html show that thing. Moreover buttons should be included algon the run and save and clear buttons, there are already many buttons in the panel heading. @championswimmer I have included it below editor.

image

**Comment by astu9880** [comment]
@hereisnaman  "close custom input" sounds too spoon fed. 
I believe as **once the language theme file is set**, the user will interact most with the **run save area** only and so it should be **kept as clean as possible**.

**Comment by astu9880** [comment]
@hereisnaman  not to mention the custom input toggle is not as frequently used a button. It is specific to the code you're writing.

**Comment by hereisnaman** [comment]
@astu9880 thanks for the suggestion. @championswimmer added close icon in input panel heading which closes the input box. 
screen shot 2017-05-28 at 2 26 03 am
screen shot 2017-05-28 at 2 26 08 am

**Comment by hereisnaman** [comment]
@championswimmer sir @astu9880 is clearly copying icon feature I already added in my PR.

**Comment by astu9880** [comment]
@hereisnaman I'd use my suggestion for my PR if that's okay. Also thanks for the picture. Definitely shows that icon and "Input" are out of line.
@championswimmer I didn't copy. Just gave icon a top margin :)

**Comment by hereisnaman** [comment]
@astu9880 those are clearly not suggestions for you. I quoted @championswimmer for it. Please re-read.

**Comment by championswimmer** [comment]
please fix the conflicts

**Comment by championswimmer** [comment]
The tone of discussion going on between @hereisnaman and @astu9880 is just unhealthy.
Please refrain from that.

I will be going for #52 over #51 because as per the commit timestamps, @astu9880 did it before @hereisnaman

NOTE: If you cannot fix the conflicts in a reasonable time - within a day or two (and @hereisnaman fixes conflicts in his PR), I might merge his.

**Comment by astu9880** [comment]
@championswimmer Merge conflicts resolved and css break point fixed. Please review.

**Comment by astu9880** [comment]
@championswimmer  Sir please take a look and close this :+1:

**Comment by astu9880** [comment]
@championswimmer the custom input is meaningless in full screen mode. So the last commit just hides the custom input button when the editor is full screen

**Comment by astu9880** [comment]
@championswimmer @Prajjwal  @prateek27 This issue is still open. Please suggest changes if you'd like any?

File: scripts/main.js [hunk 1]
```diff
@@ -158,22 +158,28 @@ $(document).ready(function () {
...
-    var fs = false;
-    $("#panel-fullscreen").click(function (e) {
+
+   var fs=false;
+   window.isInpBoxHidden=false;
+   $("#panel-fullscreen").click(function (e) {
         e.preventDefault();
 
-        fs = !fs;
+        fs=!fs;
```
Comment (omerjerk) [inline scripts/main.js]: Why did you do this?
Refrain from making changes to any of the code which your pull request isn't concerned with.

File: scripts/main.js [hunk 2]
```diff
@@ -158,22 +158,28 @@ $(document).ready(function () {
         $('#editor').css({'height': cw + 'px'});
     }
     //Toggle fullscreen
-    var fs = false;
-    $("#panel-fullscreen").click(function (e) {
+
+   var fs=false;
```
Comment (omerjerk) [inline scripts/main.js]: Why did you remove spaces around '=' ?

File: scripts/main.js [hunk 3]
```diff
@@ -158,22 +158,28 @@ $(document).ready(function () {
...
+   window.isInpBoxHidden=false;
```
Comment (omerjerk) [inline scripts/main.js]: Please add spaces around '='

File: scripts/main.js [hunk 4]
```diff
@@ -158,22 +158,28 @@ $(document).ready(function () {
...
             exitFullScreen();
         }
         var $this = $(this);
 
-        if ($this.children('i').hasClass('glyphicon-resize-full')) {
-            $this.attr('title', 'Exit Full Screen');
-            $this.children('i').removeClass('glyphicon-resize-full');
+        if ($this.children('i').hasClass('glyphicon-resize-full')){
+			      $this.attr('title','Exit Full Screen');
+			      $this.children('i').removeClass('glyphicon-resize-full');
```
Comment (omerjerk) [inline scripts/main.js]: Why all these whitespaces changes?
Your diff should be as clean as possible.

File: scripts/main.js [hunk 5]
```diff
@@ -217,4 +223,16 @@ function exitFullScreen() {
         document.mozCancelFullScreen();
     else if (document.webkitExitFullscreen)
         document.webkitExitFullscreen();
+}
+
+function toggleCustInp() {
+    window.isInpBoxHidden=!window.isInpBoxHidden;
```
Comment (omerjerk) [inline scripts/main.js]: spaces around '=' please.

File: scripts/main.js [hunk 6]
```diff
@@ -217,4 +223,16 @@ function exitFullScreen() {
...
+    var inp=$('#test-input');
```
Comment (omerjerk) [inline scripts/main.js]: same.

File: scripts/main.js [hunk 7]
```diff
@@ -217,4 +223,16 @@ function exitFullScreen() {
...
+    if(isInpBoxHidden){
+        localStorage.setItem('inputData',inp.html());
+        inp.html(null);
+    }else{
```
Comment (omerjerk) [inline scripts/main.js]: spaces.

**Comment by astu9880** [comment]
@omerjerk Fixed requested changes. Thanks! I'd put more effort in maintaining the quality. Please review

**Comment by omerjerk** [comment]
@astu9880 This looks good. I will test it once and merge.

**Comment by astu9880** [comment]
@omerjerk Okay!
